hudi-flink/src/main/java/org/apache/hudi/operator/StreamWriteFunction.java

```diff
@@ -252,6 +270,41 @@ private void initWriteFunction() {
     }
   }
 
+  /**
+   * Tool to detect if to flush out the existing buffer.
+   * Sampling the record to compute the size with 0.01 percentage.
+   */
+  private static class BufferSizeDetector {
+    private final Random random = new Random(47);
+
+    private final double batchSizeBytes;
+
+    private long lastRecordSize = -1L;
+    private long totalSize = 0L;
+
+    BufferSizeDetector(double batchSizeMb) {
+      this.batchSizeBytes = batchSizeMb * 1024 * 1024;
```

Comment:
We'd better do some legality verification, considering the config option does not hint unit.

Reply:
The configuration would check it as a valid integer, rename the config option to `write.batch.size.MB`

Reply:
It seems Flink may have gone the opposite way (removed the unit from the config name)? https://ci.apache.org/projects/flink/flink-docs-master/deployment/config.html#taskmanager-memory-jvm-metaspace-size

Reply:
I'm not planning to support unit in the config option value, so a key with unit is more clear.

Reply:
OK, it's not a key issue.